File: hw/vizcaino/u3/hw24FactoryMethod/hw24FactoryMethodJava/FactoryMethod/src/main/java/ec/edu/espe/factorymethod/WinFactory.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.factorymethod;

/**
 *
 * @author Adrian Vizcaino <The-Softwarriors at ESPE>
 */
public class WinFactory extends GUIFactory {
    public Button createButton() {
        return new WinButton();
    }
    public Menu createMenu() {
        return new WinMenu();
    }
}

File: hw/palomo/u3/hw23AbstracFactory/AbstracFactory.ts/model/WinFactory.ts
import { GUIFactory } from "./GUIFactory";
import { Button } from "./Button";
import { Menu } from "./Menu";
import { WinButton } from "./WinButton";
import { WinMenu } from "./WinMenu";

export class WinFactory implements GUIFactory {
    createButton(): Button {
        return new WinButton();
    }

    createMenu(): Menu {
        return new WinMenu();
    }
}